File: java/1175. Prime Arrangements.java
class Solution {
    public int numPrimeArrangements(int n) {
        long result = 1;
        int prime = countPrimeNumber(n);
        int mod = (int)(1e9 + 7);
        for(int i = 2; i <= prime; i++){
            result = (result*i)%mod;
        }
        for(int i = n-prime; i >= 2; i--){
            result = (result*i)%mod;
        }
        
        return (int)result;
    }
    
    
    private int countPrimeNumber(int n){
        int count = 0;
        int[] numbers = new int[n+1];
        
        for(int i = 2; i <= n; i++){
            if(numbers[i] == 0){
                for(int j = i*2; j <= n; j += i){
                    numbers[j]++;
                }
            }
        }
        
        for(int i = 2; i <= n; i++){
            if(numbers[i] == 0){
                count++;
            }
        }
        return count;
    }
}

File: javascript/1175. Prime Arrangements.js
var numPrimeArrangements = function(n) {
    let modulo = 10**9 + 7;
  
    let prime = getPrimeCount(n);;
    let result = 1;

    for(let i = 2; i <= prime; i++){
        result = (result*i)%modulo;
    }
    for(let i = n-prime; i >= 2; i--){
        result = (result*i)%modulo;
    }
    return result%modulo;
};

function getPrimeCount(n){
    let numbers = new Array(n+1).fill(false);
    let prime = 0;
    for(let i = 2; i <= n; i++){
        if(numbers[i] == false){
            for(let j = 2*i; j <= n; j+=i){
                numbers[j] = true;
            }
        }
    }
    for(let i = 2; i <= n; i++){
        if(numbers[i] == false){
            prime++;
        }
    }
    return prime;
}
